### DISCUSSION TITLE
Pictures with same date/time and same GPS coordinate are not being imported correctly

### DISCUSSION THREAD

**Discussion by svanhoutte** [photoprism/photoprism#2778] [Q&A] [answered]
#### 1. What is not working as documented?

When importing pictures from a folder, if the pictures have the same dates and the same GPS coordinate, then only the 1st picture shows up in Photoprism. All the pictures are copied from the Import folder to the original ones. 
The other pictures are not showing in the search or folder. 

#### 2. How can we reproduce it?

1. Have 2 different pictures with same date and same GPS coordinate and put them in an import forlder
2. Select the import folder in library and click on import
3. Go to folder and select the folder corresponding to the date
4. Only 1 picture is there. 

I have around 20K pictures in the library only JPG. When doing the test i only use a folder with 8 pictures.

#### 3. What behavior do you expect?

To see all the pictures in the folder.

#### 4. What could be the cause of your problem?

Tried to restart the docker, without any changes, if i change the time in the picture by 1 sec then the pictures are processing correctly. There is no error message in the logs

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Test pictures bellow you can use for samples
![IMG_4987](https://user-images.githubusercontent.com/22617546/195680155-1f43e8e8-e78a-41e6-88ff-edf0a653ee34.JPG)
![IMG_4988](https://user-images.githubusercontent.com/22617546/195680163-2d574a8f-2085-4c8c-b66a-5eacbe525ab8.JPG)
![IMG_4986](https://user-images.githubusercontent.com/22617546/195680165-6d17d436-bbbe-4b40-8142-86145efc55e6.JPG)

Logs from the docker 
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=debug msg="covers: updated 0 folders [63.712372ms]"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=debug msg="removed folder:import:true:false from cache"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=info msg="copying files from 160___10"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=info msg="import: found no .ppignore file"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=debug msg="exiftool: extracting metadata from upload/160___10/IMG_4986.JPG"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=debug msg="import: created 201368c695b570496b337e813cc914be18eecbf6_exiftool.json"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=info msg="media: IMG_4986.JPG was taken at 1979-06-12 09:00:00 +0000 UTC (meta)"
photoprism_1  | time="2022-10-13T14:27:35-04:00" level=info msg="import: moving main jpg file IMG_4986.JPG to 1979/06/19790612_090000_5B6B13B4.jpg"
photoprism_1  | time="2022-10-13T14:27:37-04:00" level=debug msg="media: opened 1979/06/19790612_090000_5B6B13B4.jpg [52 MB]"
photoprism_1  | time="2022-10-13T14:27:39-04:00" level=info msg="media: created 13 thumbnails for 1979/06/19790612_090000_5B6B13B4.jpg [3.008310733s]"
photoprism_1  | time="2022-10-13T14:27:39-04:00" level=debug msg="media: created no new thumbnails for 1979/06/19790612_090000_5B6B13B4.jpg [4.187015ms]"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=info msg="media: 19790612_090000_5B6B13B4.jpg was taken at 1979-06-12 09:00:00 +0000 UTC (meta)"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="photo: changing location of 1979/06/19790612_090000_5B6B13B4 from zz to s2:11378544fa4c"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="photo: 1979/06/19790612_090000_5B6B13B4 title based on label 'Batha البطحة'"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="photo: 1979/06/19790612_090000_5B6B13B4 has new title 'Batha البطحة / Chad / 1979' [799.653µs]"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="search: updated index by photo id [2.301836ms]"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="index: updated yaml file 19790612_090000_5B6B13B4.yml"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=info msg="import: added main jpg file 1979/06/19790612_090000_5B6B13B4.jpg"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="exiftool: extracting metadata from upload/160___10/IMG_4987.JPG"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=debug msg="import: created e7e1ec5082c42c27b2b6499edc3c304346e129ee_exiftool.json"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=info msg="media: IMG_4987.JPG was taken at 1979-06-12 09:00:00 +0000 UTC (meta)"
photoprism_1  | time="2022-10-13T14:27:40-04:00" level=info msg="import: moving main jpg file IMG_4987.JPG to 1979/06/19790612_090000_3708147D.jpg"
photoprism_1  | time="2022-10-13T14:27:41-04:00" level=debug msg="media: created no new thumbnails for 1979/06/19790612_090000_3708147D.jpg [4.187439ms]"
photoprism_1  | time="2022-10-13T14:27:41-04:00" level=debug msg="media: created no new thumbnails for 1979/06/19790612_090000_3708147D.jpg [302.117µs]"
photoprism_1  | time="2022-10-13T14:27:41-04:00" level=debug msg="search: updated index by photo id [1.626814ms]"
photoprism_1  | time="2022-10-13T14:27:41-04:00" level=info msg="import: stacked main jpg file 1979/06/19790612_090000_3708147D.jpg"
photoprism_1  | time="2022-10-13T14:27:42-04:00" level=debug msg="exiftool: extracting metadata from upload/160___10/IMG_4988.JPG"
photoprism_1  | time="2022-10-13T14:27:42-04:00" level=debug msg="import: created 25583899b1c49485e8bbcec35d089db774d507f8_exiftool.json"
photoprism_1  | time="2022-10-13T14:27:42-04:00" level=info msg="media: IMG_4988.JPG was taken at 1979-06-12 09:00:00 +0000 UTC (meta)"
photoprism_1  | time="2022-10-13T14:27:42-04:00" level=info msg="import: moving main jpg file IMG_4988.JPG to 1979/06/19790612_090000_1EECCADD.jpg"
photoprism_1  | time="2022-10-13T14:27:43-04:00" level=debug msg="media: opened 1979/06/19790612_090000_1EECCADD.jpg [52 MB]"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=info msg="media: created 13 thumbnails for 1979/06/19790612_090000_1EECCADD.jpg [2.437238721s]"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=debug msg="media: created no new thumbnails for 1979/06/19790612_090000_1EECCADD.jpg [4.091812ms]"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=debug msg="search: updated index by photo id [1.797898ms]"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=info msg="import: stacked main jpg file 1979/06/19790612_090000_1EECCADD.jpg"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=debug msg="faces: found no orphan markers [29.737394ms]"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=debug msg="markers: found no invalid references [58.832467ms]"
photoprism_1  | time="2022-10-13T14:27:45-04:00" level=debug msg="markers: found no missing subjects [1.518688ms]"
photoprism_1  | time="2022-10-13T14:28:17-04:00" level=debug msg="server: GET /api/v1/config (200) [141.706244ms]"
photoprism_1  | time="2022-10-13T14:28:17-04:00" level=debug msg="faces: found no ambiguous subjects [32.412082786s]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="faces: found no clusters to be merged [72.43081ms]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="faces: skipped clustering"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="faces: found no new faces [23.452978ms]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="faces: found no unmatched markers"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="faces: updated 0 markers, recognized 0 faces, 0 unknown [274.173228ms]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="index: updating counts"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="counts: updated 1 place [39.201081ms]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="counts: updated 0 subjects [18.461043ms]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="counts: updated 0 labels [87.999896ms]"
photoprism_1  | time="2022-10-13T14:28:18-04:00" level=debug msg="moments: analyzing 17814 photos and 734 videos, with threshold 15"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="moments: 461 albums saved as yaml files"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=info msg="import completed in 50 s"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="index: updating covers"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="covers: updated 0 albums [9.663285ms]"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="covers: updated 1 folder [46.615567ms]"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="covers: updated 1 month [21.503135ms]"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="covers: updated 0 labels [84.353049ms]"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="covers: updated 0 subjects [9.024528ms]"
photoprism_1  | time="2022-10-13T14:28:26-04:00" level=debug msg="server: POST /api/v1/import/ (200) [51.283398477s]"
photoprism_1  | time="2022-10-13T14:29:19-04:00" level=debug msg="places: found 11,085 results for public:true quality:3 count:500000 [271.7486ms]"
photoprism_1  | time="2022-10-13T14:29:20-04:00" level=debug msg="server: GET /api/v1/geo?count=500000&offset=0&q=&public=true&quality=3 (200) [513.085367ms]"
photoprism_1  | time="2022-10-13T14:29:20-04:00" level=debug msg="server: GET /api/v1/t/201368c695b570496b337e813cc914be18eecbf6/8984dd16/tile_50 (200) [5.56ms]"
photoprism_1  | time="2022-10-13T14:29:34-04:00" level=debug msg="server: GET /api/v1/albums?count=168&offset=0&q=&category=&type=folder&order=newest (200) [32.523762ms]"
photoprism_1  | time="2022-10-13T14:29:34-04:00" level=debug msg="server: GET /api/v1/t/201368c695b570496b337e813cc914be18eecbf6/8984dd16/tile_500 (200) [2.708347ms]"
photoprism_1  | time="2022-10-13T14:29:37-04:00" level=debug msg="server: GET /api/v1/albums/arjp5g41raj4c7ch (200) [1.591928ms]"
photoprism_1  | time="2022-10-13T14:29:37-04:00" level=debug msg="photos: found 2 results for filter:\"path:1979/06 public:true\" path:1979/06 public:true dist:20 album:arjp5g41raj4c7ch camera:0 count:60 order:added merged:true [2.862003ms]"
photoprism_1  | time="2022-10-13T14:29:37-04:00" level=debug msg="server: GET /api/v1/photos?count=60&offset=0&album=arjp5g41raj4c7ch&filter=path:1979%2F06+public:true&merged=true&country=&camera=0&order=added&q= (200) [6.092427ms]"
photoprism_1  | time="2022-10-13T14:29:37-04:00" level=debug msg="server: GET /api/v1/t/201368c695b570496b337e813cc914be18eecbf6/8984dd16/tile_224 (200) [2.276157ms]"

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64,  Build [220901-f493607b0]

(b) Database Type & Version: MariaDB

(c) Operating System Types & Versions: Linux

(d) Browser Types & Versions: Firefox, Chrome

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?

Running Ubuntu server Ubuntu 22.04 jammy
Linux main1 5.15.0-50-generic #56-Ubuntu SMP Tue Sep 20 13:23:26 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type:Intel Xeon D-1567 @ 8x 2.1GHz

(b) Physical Memory & Swap Space in GB 16GB physical memory 4GB Swap

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,... System on SSD, pictures on NAS (~250Mb/sec)

(d) Anything else that might be helpful to know?
:~/photoprism$ free -h
               total        used        free      shared  buff/cache   available
Mem:            15Gi       3.5Gi       983Mi       4.0Mi        11Gi        12Gi
Swap:          4.0Gi       1.0Mi       4.0Gi

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

no

**Comment by graciousgrey** [comment, marked as answer]
Files with the exact same location and date+time get stacked. 
You can turn this of in the library settings. As mentioned in our first steps guide, library settings should ideally be configured before the initial indexing.

https://docs.photoprism.app/user-guide/first-steps/
https://docs.photoprism.app/user-guide/settings/library/
https://docs.photoprism.app/user-guide/organize/stacks/

You can [unstack](https://docs.photoprism.app/user-guide/organize/stacks/#unstack-files) the already indexed files manually. If you turn of stacking by date and time before indexing the rest of your library those won't be stacked.

**Reply by svanhoutte** [reply]
Thanks a lot for the answer. The least i can do is sponsor you now and it is done.
Great project looking forward to use it more. It made me want to do more photography.
